### PR TITLE
KVO Crash Remove PlayerViewController Observers on vc dismiss

### PR DESCRIPTION
**Pull request by samermurad**
The KVO crash still happens under certain circumstances, for example when having a view that still displays the video component after finishing the video.

Currently the KVO of the PlayerViewController are only being removed on removeFromSuperview.
Which problematic because the `_playerViewController` instance is being set to `nil` in the  `videoPlayerViewControllerDidDismiss`
which deallocates the the instances because it is no longer retained, so if the `removeFromSuperview` method is not called directly after that, you will get the KVO crash on the two values "frame" and "readyForDisplay" because they are both registered on the `_playerViewController` and `_playerViewController. contentOverlayView`.
this happen in situations where you are displaying the video object after the video is finished as well, for example in a listView or an article, where the user will still be on the page and the Video element is still rendered.
Video is not being played, just displayed and the `_playerViewController` was deallocated, but `removeFromSuperview` was yet to be called, the view is still visible.

Longer story short, the `_playerViewController` KVO listeners must be removed before nullifying the  `_playerViewController` object

Thanks for opening a PR!
Since this is a volunteer project and is very active, anything you can do to reduce the amount of time needed to review and merge your PR is appreciated.
The following steps will help get your PR merged quickly:

#### Update the documentation
If you've added new functionality, update the README.md with an entry for your prop or event.
The entry should be inserted in alphabetical order.

#### Update the changelog
After you open the PR, update the CHANGELOG.md file with an entry pointing to your PR.

#### Provide an example of how to test the change
If the PR requires special testing setup provide all the relevant instructions and files. This may include a sample video file or URL, configuration, or setup steps.

#### Focus the PR on only one area
Testing multiple features takes longer than isolated changes and if there is a bug in one feature, prevents the other parts of your PR from getting merged until it gets fixed.
If you're touching multiple different areas that aren't related, break the changes up into multiple PRs.

#### Describe the changes
Add a note describing what your PR does. If there is a change to the behavior of the code, explain why it needs to be updated.

### CODE REVIEW

**Review by jensellfors — APPROVED** [review]
Never had a crash for this but it all sounds correct, nice one!

File: ios/Video/RCTVideo.m [hunk 1]
```diff
@@ -530,7 +531,6 @@ - (void)playerItemForSource:(NSDictionary *)source withCallback:(void(^)(AVPlaye
 - (void)playerItemForSourceUsingCache:(NSString *)uri assetOptions:(NSDictionary *)options withCallback:(void(^)(AVPlayerItem *))handler {
     NSURL *url = [NSURL URLWithString:uri];
     [_videoCache getItemForUri:uri withCallback:^(RCTVideoCacheStatus videoCacheStatus, AVAsset * _Nullable cachedAsset) {
-        switch (videoCacheStatus) {
```
Comment (jensellfors) [inline ios/Video/RCTVideo.m]: Oh I missed this, this must be a typo no? You can't remove the switch case like that? @samermurad

**Review by jensellfors — CHANGES_REQUESTED** [review]

File: ios/Video/RCTVideo.m (hunk 1 above)
Comment (samermurad) [reply ios/Video/RCTVideo.m]: @jenshandersson oh wow yeah this is definitely a typo, I'll Fix it in a sec

Comment (samermurad) [reply ios/Video/RCTVideo.m]: @jenshandersson Fixed

**Comment by samermurad** [comment]
@jenshandersson Regarding the reproduction of the crash, I think it's a special situation where you still keep the Video Thumbnail displayed (in a list for example) After dismissing the video player The Idea is to allow the `videoPlayerViewControllerDidDismiss` to be called but still keep the view thumbnail view on screen, the crash doesn't happen instantly it takes about 1.5s sec to happen (I guess it depends on the device).
But this commit fixes it anyways so not sure the long explanation is necessary, although is probably helpful for future reference

**Review by jensellfors — APPROVED** [review]
@samermurad thanks. Can you also add a line to the Changelog please, then I'm happy to merge

**Comment by samermurad** [comment]
@jenshandersson Sorry I totally missed that message, will do so shortly

**Comment by samermurad** [comment]
@jefersondaniel (I'll fix the merge conflict as well) but under what version name should I add the changelog?

**Pull request closed without merging**